docs/reference/aggregations/misc.asciidoc

```diff
@@ -82,3 +82,83 @@ Then that piece of metadata will be returned in place for our `titles` terms agg
 // TESTRESPONSE[s/\.\.\./"took": "$body.took", "timed_out": false, "_shards": "$body._shards", "hits": "$body.hits"/]
+
+
+[[returning-aggregation-type]]
+== Returning the type of the aggregation
+
+Sometimes you need to know the exact type of an aggregation in order to parse its results. The `prefixed_name` parameter
+ can be used to change the aggregation's name in the response so that it will be prefixed by its internal type.
+
+Considering the following <<search-aggregations-bucket-datehistogram-aggregation,`date_histogram` aggregation>> named
+`tweets_over_time` which has a sub <<search-aggregations-metrics-top-hits-aggregation, 'top_hits` aggregation>> named
+ `top_users`:
+
+[source,js]
+--------------------------------------------------
+GET /twitter/tweet/_search?prefixed_name
+{
+  "aggregations": {
+    "tweets_over_time": {
+      "date_histogram": {
+        "field": "date",
+        "interval": "year"
+      },
+      "aggregations": {
+        "top_users": {
+            "top_hits": {
+                "size": 1
+            }
+        }
+      }
+    }
+  }
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[setup:twitter]
+
+In the response, the aggregations names will be changed to respectively `date_histogram:tweets_over_time` and
+`top_hits:top_users`, reflecting the internal types of each aggregation:
+
+[source,js]
+--------------------------------------------------
+{
+    "aggregations": {
+        "date_histogram:tweets_over_time": { <1>
+            "buckets" : [
+                {
+                    "key_as_string" : "2009-01-01T00:00:00.000Z",
+                    "key" : 1230768000000,
+                    "doc_count" : 5,
+                    "top_hits:top_users" : {  <2>
+                        "hits" : {
+                            "total" : 5,
+                            "max_score" : 1.0,
+                            "hits" : [
+                                {
+                                  "_index": "twitter",
+                                  "_type": "tweet",
+                                  "_id": "0",
+                                  "_score": 1.0,
+                                  "_source": {
+                                    "date": "2009-11-15T14:12:12",
+                                    "message": "trying out Elasticsearch",
+                                    "user": "kimchy",
+                                    "likes": 0
+                                  }
+                                }
+                            ]
+                        }
+                    }
+                }
+            ]
+        }
+    },
+    ...
+}
+--------------------------------------------------
+// TESTRESPONSE[s/\.\.\./"took": "$body.took", "timed_out": false, "_shards": "$body._shards", "hits": "$body.hits"/]
+
+<1> The name `tweets_over_time` now contains the `date_histogram` prefix.
+<2> The name `top_users` now contains the `top_hits` prefix.
```

Comment:
shall we document that in certain cases the returned type is not exactly the type that was provided in the request? I think this happens only for terms and significant terms.

Also, I wonder if we should avoid returning umterms and umsigterms, as whether the field is unmapped should not be relevant, rather return sterms and ssigterms for both. We could also potentially share the toXContet and parsing code in `UnmappedTerms` and `UnmappedSignificantTerms` with `StringTerms` and `SignificantStringTerms`.